### PR TITLE
add markdownlint to ci

### LINKED ISSUE #119: Add markdown linter in CI
It would be nice to have something in CI that tells us if markdown is correct.
For example this markdown is not valid:

```
# section 1

### section 2
```

Or stuff like: avoid tabs or detect trailing whitespaces.

### PR DESCRIPTION
**Pull request by Takashiidobe**
This PR adds markdownlint to CI.

Fixes #119

### CODE REVIEW

**Comment by Takashiidobe** [comment]
Once PR #163 is approved, i'll apply those changes to this PR, and that should allow it to pass CI.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,3 +41,17 @@ jobs:
...
 
       - run: mdbook test
+  lint:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Lint all files in current directory
+        uses: avto-dev/markdown-lint@v1
+        with:
+          config: '.markdownlint.json'
```
Comment (simonsan) [inline .github/workflows/ci.yml:51]:
```suggestion
          config: '.markdownlint.yaml' 
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -41,3 +41,17 @@ jobs:
...
+      - name: Lint all files in current directory
+        uses: avto-dev/markdown-lint@v1
+        with:
+          config: '.markdownlint.json' 
+          args: '*.md'
+      - name: Lint all files recursively
+        uses: avto-dev/markdown-lint@v1
+        with:
+          config: '.markdownlint.json' 
+          args: '**/*.md'
```
Comment (simonsan) [inline .github/workflows/ci.yml:57]:
```suggestion
          config: '.markdownlint.yaml' 
          args: '**/*.md'

```

**Comment by simonsan** [comment]
Nice thank you! :-)

**Pull request merged**